File: frontend/edenBooks/src/modules/messages/services/messageService.ts
import { IMessageService } from "../interfaces/IMessageService";
import { messageApi } from "../api/messageApi";
import { IMessageRequest } from "../interfaces/IMessageRequest";

export const messageService: IMessageService = {
  CreateMessage: async (message: IMessageRequest) =>messageApi.CreateMessage(message),
  GetMessages: async () => messageApi.GetMessages(),
  GetMessageById: async (id: number) => messageApi.GetMessageById(id),
  UpdateMessage: async (id: number, message: IMessageRequest) =>messageApi.UpdateMessage(id, message),
  DeleteMessage: async (id: number) => messageApi.DeleteMessage(id),
};

File: backend/internal/modules/messages/domain/services/messageService.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/messages/application/dto"
	"github.com/javierjpv/edenBooks/internal/modules/messages/domain/entities"
)

type MessageService interface {
	CreateMessage(message dto.MessageRequest) (*entities.Message, error)

	UpdateMessage(id uint, message dto.MessageRequest) error

	DeleteMessage(id uint) error

	GetMessageByID(id uint) (*entities.Message, error)

	GetFilteredMessages(filters map[string]string) ([]entities.Message, error)
}
